ddtrace/tracer/span.go

```diff
@@ -337,6 +337,24 @@ func (s *span) String() string {
 	return strings.Join(lines, "\n")
 }
 
+// Format implements fmt.Formatter.
+func (s *span) Format(f fmt.State, c rune) {
+	switch c {
+	case 's':
+		fmt.Fprint(f, s.String())
+	case 'v':
+		fmt.Fprintf(f, "dd.trace_id=%d dd.span_id=%d dd.service=%s", s.TraceID, s.SpanID, s.Service)
```

Comment:
Let's find the cleanest and least intrusive way to get the right service here.

Reply:
`globalconfig.ServiceName()`  is not set unless the user explicitly sets a service name. Unfortunately, I think we still need to use `tracer.config.Service` here to avoid empty service names when the user doesn't set a service name and leaves it as the name of the program:
https://github.com/DataDog/dd-trace-go/blob/435545bd358b5032d8e6e66cf1cab16c6a712a4c/ddtrace/tracer/option.go#L114-L116

Reply:
We should change the code so that `globalconfig.ServiceName` is always set. In this case, we would simply set it to that default. What do you think?

Reply:
> globalconfig.ServiceName() is not set unless the user explicitly sets a service name.

Turns out, this is what we want.